rule/string-regex.go

```diff
@@ -0,0 +1,233 @@
+	Regexp       *regexp.Regexp
+	ErrorMessage string
+}
+
+type stringRegexSubruleScope struct {
+	Func     string // Function name the rule is scoped to
+	Argument int    // (optional) Which argument in calls to the function is checked against the rule (the first argument is checked by default)
+	Field    string // (optional) If the argument to be checked is a struct, which member of the struct is checked against the rule (top level members only)
+}
+
+var parseStringRegexScope = regexp.MustCompile("^([A-Za-z][A-Za-z0-9]+)(?:\\[([0-9]+)\\])?(?:\\.([A-Za-z]+))?$")
+
+// #endregion
+
+// #region Argument parsing
+
+func (w *lintStringRegexRule) parseArguments(arguments lint.Arguments) {
+	for i, argument := range arguments {
+		scope, regex, errorMessage := w.parseArgument(argument, i)
+		w.rules = append(w.rules, stringRegexSubrule{
+			Parent:       w,
+			Scope:        scope,
+			Regexp:       regex,
+			ErrorMessage: errorMessage,
+		})
+	}
+}
+
+func (w lintStringRegexRule) parseArgument(argument interface{}, ruleNum int) (scope stringRegexSubruleScope, regex *regexp.Regexp, errorMessage string) {
+	g, ok := argument.([]interface{}) // Cast to generic slice first
+	if !ok {
+		panic(fmt.Sprintf("unable to parse argument %d", ruleNum))
+	}
+	var rule []string
+	for i, obj := range g {
+		val, ok := obj.(string)
+		if !ok {
+			panic(fmt.Sprintf("unable to parse argument %d, option %d", ruleNum, i))
+		}
+		rule = append(rule, val)
+	}
+
+	// Parse rule scope
+	scope = stringRegexSubruleScope{}
+	matches := parseStringRegexScope.FindStringSubmatch(rule[0])
+	if matches == nil {
+		panic(fmt.Sprintf("unable to parse rule scope (argument %d, option 0)", ruleNum))
+	}
+	scope.Func = matches[1]
+	if len(matches[2]) > 0 {
+		var err error
+		scope.Argument, err = strconv.Atoi(matches[2])
+		if err != nil {
+			fmt.Println(len(matches))
+			panic(fmt.Sprintf("unable to parse rule scope argument number (argument %d, option 0)", ruleNum))
+		}
+	}
+	if len(matches[3]) > 0 {
+		scope.Field = matches[3]
+	}
+
+	// Strip / characters from the beginning and end of rule[1] before compiling
+	regex, err := regexp.Compile(rule[1][1 : len(rule[1])-1])
+	if err != nil {
+		panic(fmt.Sprintf("unable to compile %s as regexp (argument %d, option 1)", rule[1], ruleNum))
+	}
+
+	// Use custom error message if provided
+	if len(rule) == 3 {
+		errorMessage = rule[2]
+	}
+	return scope, regex, errorMessage
+}
+
+// #endregion
+
+// #region Node traversal
+
+func (w lintStringRegexRule) Visit(node ast.Node) ast.Visitor {
+	// First, check if node is a call expression
+	call, ok := node.(*ast.CallExpr)
+	if !ok {
+		return w
+	}
+
+	// Get the name of the call expression to check against rule scope
+	callName, ok := w.getCallName(call)
+	if !ok {
+		return w
+	}
+
+	for _, rule := range w.rules {
+		if rule.Scope.Func == callName {
+			rule.Apply(call)
+		}
+	}
+
+	return w
+}
+
+// Return the name of a call expression in the form of package.Func or Func
```

Comment:
Just an escape route in case both casts fail. This shouldn't happen in practice, but I try to never perform casts in Go without checking for success, which would increase the overall risk of undefined/unexpected behavior.